pkg/start/start.go

```diff
@@ -417,6 +417,27 @@ func useProtobuf(config *rest.Config) {
 	config.ContentType = "application/vnd.kubernetes.protobuf"
 }
 
+func getLeaderElectionConfig(ctx context.Context, restcfg *rest.Config) configv1.LeaderElection {
+
+	// Defaults follow conventions
+	// https://github.com/openshift/enhancements/pull/832/files#diff-2e28754e69aa417e5b6d89e99e42f05bfb6330800fa823753383db1d170fbc2fR183
+	// see rhbz#1986477 for more detail
+	defaultLeaderElection := libgoleaderelection.LeaderElectionDefaulting(
+		configv1.LeaderElection{},
+		"", "",
+	)
+
+	if infra, err := clusterstatus.GetClusterInfraStatus(ctx, restcfg); err == nil && infra != nil {
+		if infra.ControlPlaneTopology == configv1.SingleReplicaTopologyMode {
+			return libgoleaderelection.LeaderElectionSNOConfig(defaultLeaderElection)
```

Comment:
We're not all that particular about our lease settings, and we do have reasonably solid graceful shutdown logic where we release the leases when termination gives us enough time to do that.  So I'm personally fine simplifying all of this and just using `libgoleaderelection.LeaderElectionSNOConfig` all the time, regardless of the Infrastructure configuration.  Thoughts?